sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -283,8 +292,12 @@ public String getString(int idx) {
    */
   @Nullable
   public ReadableDateTime getDateTime(int idx) {
-    ReadableInstant instant = getValue(idx);
-    return instant == null ? null : new DateTime(instant).withZone(instant.getZone());
+    Instant instant = getValue(idx);
```

Comment:
I think `public abstract <T> T getValue(int fieldIdx);` should be changed to `private` and all value access to Row should go though specific type access method, e.g. `public int16 getInt6(int idx)`, `public ReadableDateTime getDateTime`, etc. 

By doing so, we can hide the internal representation of Row. 

Right now I at least see BeamSQL implementation uses `public <T> T getValue(int fieldIdx)` directly, which access the value saved in Row.